src/meson.build

```diff
@@ -102,10 +101,19 @@ fwupdutil = library(
   ],
 )
 
-if build_daemon
-install_data(['org.freedesktop.fwupd.xml'],
-  install_dir: join_paths(datadir, 'dbus-1', 'interfaces')
+dbus_interface = custom_target('fwupd-generate-dbus-interface',
+  input : 'org.freedesktop.fwupd.xml',
+  output : 'org.freedesktop.fwupd.xml',
+  command : [
+    join_paths(meson.project_source_root(), 'contrib', 'generate-dbus-interface.py'),
```

Comment:
This is assuming that the shebang in the top works which I thought is a no-no from past issues.  So aren't you missing a python launcher for this?

Reply:
Also - we shouldn't be putting python scripts that are mandatory for the build process into `contrib`.

Reply:
>  So aren't you missing a python launcher for this?

100% yes, good catch.

Reply:
> python scripts that are mandatory for the build

I couldn't see a better option, and I think we need to move quite a few around now. libfwupd or libfwupdplugin don't feel right -- maybe `data`?

Reply:
not `build` -- maybe a `meson-helpers` toplevel directory?

Reply:
the problem with `src` is that it's visited _late_ in the build process; if we had something visited before libfwupd we could just do a `mesonhelperdir = dir(''.')` and then refer to the generate python scripts with `file`

Reply:
Okay, fixed.